### ISSUE TITLE
Upload and Integration of Historic Releases

### ISSUE THREAD

**Issue by bmcollier** [bmcollier/beebem#10]
This is a comment thread to discuss the collection of earlier versions of BeebEm for various platforms.

I had started with 0.0.13 for Linux, and @irrelevantdotcom has uploaded the following to Dropbox:

1.0
1.41
1.50
1.60
2.1a
3.2
3.85
4.0
4.01
4.02
4.11
4.12
4.14

I would like to propose creating release branches for each one, so for instance, the Windows version of v.3.85 would be branch "rel-3.85-win". Any objections?

The new version will go in a branch called "develop" which will probably be forked from Windows version 4.14, as most of the code I'm being sent seems to be extensions of the later Windows version. If that gets converted to, eg. Qt4, then it would be a good starting point for a cross-platform release.

**Comment by fesh0r** [comment]
I have a fairly large archive of beebem versions for the various platforms, which i used to create https://github.com/fesh0r/beebem
Looking at the dropbox link there are a few provided that I hadn't tracked down previously.

A txz of everything i have is at [beebem-full.tar.xz](https://www.dropbox.com/s/uhf6xrhkdfg5a4k/beebem-full.tar.xz?dl=0)
And a list of the various versions and dates [beebem all versions.txt](https://www.dropbox.com/s/l0tvfmol888kwr7/beebem%20all%20versions.txt?dl=0)

**Comment by bmcollier** [comment]
Brilliant, thanks. I've added you as a contributor.

**Comment by bmcollier** [comment]
Quick update on this one - I've added the 4.14 release for windows as a branch, will do the others later.

So, in terms of branch activity - I would like to propose:

rel-0.0.13-linux: Last working Linux version
rel-4.14-windows: Latest Windows version
develop: Based on rel-0.0.13-linux: We'll backport all the existing features from the latest Windows version, file by file, and slot in Qt4 where possible. This version should compile on Windows and Linux as soon as possible.
master: From now on, will be updated each time we make a release.
New rel-branches: Suggest we go with rel-0.0.14a as soon as we're compiling on Windows and Linux. Once we're feature complete, we'll move to rel-5.0.0.

**Comment by jgharston** [comment]
Ben Collier wrote:
> So, in terms of branch activity - I would like to propose:
...
>  New rel-branches: Suggest we go with rel-0.0.14a as soon as we're
> compiling on Windows and Linux. Once we're feature complete, we'll
> move to rel-5.0.0.

When I've been in this position I've tended to go to, eg, version 4.2x
when consolidating the various branches before moving up to 5.xx

-- 
J.G.Harston - jgh@mdfs.net - mdfs.net/jgh

**Comment by bmcollier** [comment]
Yeah, would make sense. My only concern is confusing people who might find less functionality in the early cross-platform releases. Could mark them alpha or dev-only releases, I suppose: rel-4.2.1a etc.

**Comment by eelcohuininga** [comment]
Release 4.2.1 alpha sounds reasonable to me if you'd want to move to 5.0.0 in the end

**Comment by bmcollier** [comment]
I'll create a series of issues tagged "cross-platform integration" to cover each of the source files across versions.

**Comment by jgharston** [comment]
eelcohuininga wrote:
> Release 4.2.1 alpha sounds reasonable to me if you'd want to move to
> 5.0.0 in the end

What I tend to do is something like:
4.21dev Merger of Win4.14 and Mac4.08
4.22dev Merged with Linux4.10
4.23dev Bugfix caused by conflict with Mac4.08 and Linux4.10
4.24dev Merged with Android 3.82
etc

-- 
J.G.Harston - jgh@mdfs.net - mdfs.net/jgh

**Comment by bmcollier** [comment]
Gotcha. I've made the next milestone 4.21a, and updated the issues accordingly.

**Comment by irrelevantdotcom** [comment]
Mac version is at http://www.g7jjf.com/download.htm
I have the source in a zip, [https://www.dropbox.com/s/y351j22v4fx16nd/MacBeebEmSrc33a.zip?dl=0](url)
